src/what-unsafe-does.md

```diff
@@ -16,18 +16,35 @@ to your program. You definitely *should not* invoke Undefined Behavior.
 Unlike C, Undefined Behavior is pretty limited in scope in Rust. All the core
 language cares about is preventing the following things:
 
-* Dereferencing null, dangling, or unaligned pointers
+* Dereferencing (using the `*` operator on) null, dangling, or unaligned
+  pointers
 * Reading [uninitialized memory][]
 * Breaking the [pointer aliasing rules][]
-* Producing invalid primitive values:
-    * dangling/null references
-    * null `fn` pointers
+* Producing invalid primitive values (either alone or as a field of a compound
+  type such as `enum`/`struct`/array/tuple):
     * a `bool` that isn't 0 or 1
     * an undefined `enum` discriminant
+    * null `fn` pointers
     * a `char` outside the ranges [0x0, 0xD7FF] and [0xE000, 0x10FFFF]
-    * A non-utf8 `str`
+    * a `!` (all values are invalid for this type)
+    * dangling/null/unaligned references, references that do themselves point to
+      invalid values, or fat references (to a dynamically sized type) with
+      invalid metadata
+    * a non-utf8 `str`
+    * an uninitialized integer (`i*`/`u*`) or floating point value (`f*`)
+    * an invalid library type with custom invalid values, such as a `NonNull` or
+      `NonZero*` that is 0
 * Unwinding into another language
 * Causing a [data race][race]
+* Executing code compiled with platform features that the current platform does
```

Comment:
```suggestion
* Executing code compiled with target-features that the current thread of execution does
```

These are historically-called target-features but nowadays targets do support different features in different threads of execution, e.g., on heterogeneous SoCs where one core supports features that the others do not.